src/box/lua/config/instance_config.lua

```diff
@@ -1463,7 +1463,7 @@ return schema.new('instance_config', schema.record({
         synchro_timeout = schema.scalar({
             type = 'number',
             box_cfg = 'replication_synchro_timeout',
-            default = 5,
+            default = 0,
```

Comment:
The value should depend on compat value. Like it's done with `box_cfg_replication_synchro_timeout`

Reply:
Andrey Saranchin said that the way it was done with `box_cfg_replication_sync_timeout` is wrong. This does not work correctly because [this code](https://github.com/tarantool/tarantool/blob/9d3859b246977c307a22f5930755fe77e374a634/src/box/lua/config/instance_config.lua#L1473) is executed first, and only then the compat options are set.

He also said that if the option is not expressed declaratively (for example, if its value depends on the compat option), `box.NULL` should be set here. Just like this is done, for example, with `election_mode`:
https://github.com/tarantool/tarantool/blob/9d3859b246977c307a22f5930755fe77e374a634/src/box/lua/config/instance_config.lua#L1491-L1501
In this case, the option value will be pulled from table `default_cfg` (load_cfg.lua).

Reply:
Fixed (`box.NULL` is set).

Reply:
Created ticket for that: https://github.com/tarantool/tarantool/issues/10280. Thank you for investigation, now it works correctly, tested manually